test/tests/functional/pbs_power_provisioning_sgi.py

```diff
@@ -38,6 +38,7 @@
 from tests.functional import *
 
 
+@requirements(no_mom_on_server=True)
```

Comment:
Is this really necessary? The test suite's custom setUp() deletes any mom on the server if present.

Reply:
@agrawalravi90 Yes it is necessary because there are external tools which uses this information to create PBS cluster while running these tests... + Purpose of `@requirements` is to provide information to such external tools and skip tests if requirements don't match.

If you think this is not right then I suggest you to start discussion whatever you think but for now, I think we should allow this PR.

For more detail on requirements decorator please have a look at https://pbspro.atlassian.net/wiki/spaces/PD/pages/458883073/PP-1281+New+decorator+in+PTL+using+which+user+can+provide+cluster+information+required+for+a+test.

Reply:
So, allowing this PR will mean that I as a developer will not be able to run this test on my dev system unless I first delete the mom on my system before running this test, otherwise the requirements decorator will skip it, right? As of now, the test itself does this for me, just like all other PTL tests which create/delete various PBS daemons. That seems undesirable to me.

If we need a decorator to convey information to an external tool then why don't we create one which will not skip the test ? or add a skip argument to the same decorator which can be set to "False".

Reply:
@agrawalravi90 TestSuite's setup is run after decorators are applied. If we add skip argument to existing decorator then TH2 will not be able to find if it is single node test or multi-node test.

Reply:
Thanks for your inputs @shilpakodli . I'm actually not aware of how TH2 works, we can brainstorm offline what's the best way to solve this since TH2 is Altair's internal software. But as long as this change affects PTL testing negatively/doesn't make sense as far as the open source version is concerned, I cannot allow it.

Reply:
As per my understanding of the document - we compare the requirements flags with what is passed to pbs_benchpress command. I do not think we would need to delete the mom on the server host for the test/testsuite to not be skipped. If we do need to delete the mom, the design needs to be clear about that.
If my understanding is correct - we can allow this PR to go ahead.

Reply:
@agrawalravi90 : The test is specific to SGI platforms. To simulate the SGI cluster it expects no mom on server node. More ever this test doesn't do much of non SGI platforms. This PR is about adding test requirements using new decorator, so that external tools can create cluster and run the test cases.   
May be @ashwathraop  can add more details.

Reply:
> As per my understanding of the document - we compare the requirements flags with what is passed to pbs_benchpress command. I do not think we would need to delete the mom on the server host for the test/testsuite to not be skipped. If we do need to delete the mom, the design needs to be clear about that.
> If my understanding is correct - we can allow this PR to go ahead.

I agree.

Reply:
> So, allowing this PR will mean that I as a developer will not be able to run this test on my dev system unless I first delete the mom on my system before running this test, otherwise the requirements decorator will skip it, right?

Well, this depends on what you pass in -p in pbs_benchpress if you don't pass nomom=<sever node> while running pbs_benchpress then yes requirements decorator will skip it and that's exectly what we want. We don't want to run this kind a test which is meant to run with only remote mom.

> If we need a decorator to convey information to an external tool then why don't we create one which will not skip the test ? or add a skip argument to the same decorator which can be set to "False".

As I said in an earlier comment, if you think we need to change requirements decorator then we can discuss it but blocking this PR is not the right thing to do.

> But as long as this change affects PTL testing negatively/doesn't make sense as far as the open source version is concerned, I cannot allow it.

Can you please elaborate more on how this change is affecting PTL testing negatively?

Reply:
> I do not think we would need to delete the mom on the server host for the test/testsuite to not be skipped. If we do need to delete the mom, the design needs to be clear about that.

@prakashcv13 You are absolutely right, we don't need to delete or do anything to skip test. It should be handled by requirements decorator which it is doing... aka skip test if mom is specified on the server node.